invoices/update.go

```diff
@@ -206,15 +229,91 @@ func updateMpp(ctx *invoiceUpdateCtx,
 		return &update, ctx.acceptRes(resultAccepted), nil
 	}
 
+	var (
+		htlcPreimages map[channeldb.CircuitKey]lntypes.Preimage
+		htlcPreimage  lntypes.Preimage
+	)
+	if ctx.amp != nil {
```

Comment:
Perhaps this should be split out to a new method since this will never run for a pure MPP payment? 

It may also make the individual components easier to test as well.

Reply:
Can also serve to reduce vertical distance and nesting here a bit

Reply:
done